buildscripts/generate-manifests.sh

```diff
@@ -1,44 +1,79 @@
 #!/usr/bin/env bash
-
 set -o errexit
 set -o nounset
 set -o pipefail
 
-SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]:-"$0"}")")"
-ROOT_DIR="$SCRIPT_DIR/.."
-DEPLOY_YAML_DIR="$ROOT_DIR/deploy/yamls"
-HELM_CHART_DIR="$ROOT_DIR/deploy/helm/charts/"
-CRD_CHART_TEMPLATE_DIR="$HELM_CHART_DIR/charts/crds/templates"
-CONTROLLER_GEN=$(which controller-gen)
+# Make unmatched globs expand to nothing (prevents mv errors when no matches)
+shopt -s nullglob
+
+# Zero-width space (U+200B) sanitizer
+ZWSP=$'\u200b'
+sanitize() { printf '%s' "${1//$ZWSP/}"; }
+
+# Resolve important paths and sanitize any hidden U+200B characters
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]:-"$0"}")" && pwd -P)"
+SCRIPT_DIR="$(sanitize "${SCRIPT_DIR}")"
+ROOT_DIR="$(sanitize "${SCRIPT_DIR}/..")"
+DEPLOY_YAML_DIR="$(sanitize "${ROOT_DIR}/deploy/yamls")"
+HELM_CHART_DIR="$(sanitize "${ROOT_DIR}/deploy/helm/charts")"
+CRD_CHART_TEMPLATE_DIR="$(sanitize "${HELM_CHART_DIR}/charts/crds/templates")"
+
 RELEASE_NAME="openebs"
 RELEASE_NAMESPACE="kube-system"
 
-if [ "$CONTROLLER_GEN" = "" ]; then
-  echo "ERROR: failed to get controller-gen, Please run make bootstrap to install it";
-  exit 1;
+CONTROLLER_GEN="$(command -v controller-gen || true)"
+HELM_BIN="$(command -v helm || true)"
+
+if [[ -z "${CONTROLLER_GEN}" ]]; then
+  echo "ERROR: controller-gen not found. Please run 'make bootstrap' to install it." >&2
+  exit 1
+fi
+
+if [[ -z "${HELM_BIN}" ]]; then
+  echo "ERROR: helm not found. Please install Helm (https://helm.sh)." >&2
+  exit 1
+fi
+
+# Ensure output directories exist
+mkdir -p -- "${DEPLOY_YAML_DIR}" "${CRD_CHART_TEMPLATE_DIR}"
+
+# If a stray 'yamls' directory with a trailing U+200B exists, consolidate and remove it
+if [[ -d "${DEPLOY_YAML_DIR}${ZWSP}" ]]; then
+  shopt -s dotglob nullglob
```

Comment:
do we need to revert this and end of the if?

Reply:
Indeed. Done!